core/src/main/scala/flow/FlowResult.scala

```diff
@@ -11,16 +11,30 @@ import scala.collection.immutable.{Iterable, Seq}
   * to be propogated.
   */
 object FlowResult {
+  def apply[T, Ctx](
```

Comment:
Most changes in this file is to move FlowSuccess and FlowFailure into their own case classes that extend FlowResult, which is now a sealed trait